boilerplate/flyte/end2end/test_run.py

```diff
@@ -210,75 +205,12 @@ def run(
             "color": background_color,
         }
         results.append(result)
-    return results
-
-
-@click.command()
-@click.argument("flytesnacks_release_tag")
-@click.argument("priorities")
-@click.argument("config_file")
-@click.option(
-    "--return_non_zero_on_failure",
-    default=False,
-    is_flag=True,
-    help="Return a non-zero exit status if any workflow fails",
-)
-@click.option(
-    "--terminate_workflow_on_failure",
-    default=False,
-    is_flag=True,
-    help="Abort failing workflows upon exit",
-)
-@click.option(
-    "--test_project_name",
-    default="flytesnacks",
-    type=str,
-    is_flag=False,
-    help="Name of project to run functional tests on",
-)
-@click.option(
-    "--test_project_domain",
-    default="development",
-    type=str,
-    is_flag=False,
-    help="Name of domain in project to run functional tests on",
-)
-@click.argument(
-    "cluster_pool_name",
-    required=False,
-    type=str,
-    default=None,
-)
-def cli(
-    flytesnacks_release_tag,
-    priorities,
-    config_file,
-    return_non_zero_on_failure,
-    terminate_workflow_on_failure,
-    test_project_name,
-    test_project_domain,
-    cluster_pool_name,
-):
-    print(f"return_non_zero_on_failure={return_non_zero_on_failure}")
-    results = run(
-        flytesnacks_release_tag,
-        priorities,
-        config_file,
-        terminate_workflow_on_failure,
-        test_project_name,
-        test_project_domain,
-        cluster_pool_name,
-    )
 
-    # Write a json object in its own line describing the result of this run to stdout
     print(f"Result of run:\n{json.dumps(results)}")
 
-    # Return a non-zero exit code if core fails
     if return_non_zero_on_failure:
         for result in results:
             if result["status"] not in ("passing", "coming soon"):
-                sys.exit(1)
-
+                pytest.fail("Workflow execution failed")
 
-if __name__ == "__main__":
-    cli()
+    return results 
```

Comment:
Tests shouldn't really return anything. Can you instead assert on `results`?

Reply:
Okay ! Thank you for your review ~